backend/src/main/java/com/example/subjecthub/controller/UniversityController.java

```diff
@@ -0,0 +1,38 @@
+package com.example.subjecthub.controller;
+
+import com.example.subjecthub.api.UniversityServiceApi;
+import com.example.subjecthub.entity.University;
+import com.example.subjecthub.repository.UniversityRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.List;
+
+@RestController
+@ParametersAreNonnullByDefault
+@CrossOrigin(origins = "http://localhost:4200")
+public class UniversityController implements UniversityServiceApi {
+
+    @Autowired
+    private UniversityRepository universityRepository;
+
+    @Override
+    @RequestMapping(value = "", method = RequestMethod.GET)
+    public List<University> getUniversities(
+        @RequestParam(required = false) String abbreviation,
+        @RequestParam(required = false) String name
+    ) {
+        if (abbreviation != null && name != null)
```

Comment:
This method needs to handle 4 possibilities
1. abbreviation and name are not supplied, so return all
2. abbreviation is supplied, so return all with the matching abbreviation
3. name is supplied, so return all with the matching name
4. abbreviation and name are supplied, so return subjects that match both.

Currently, only if both are supplied you are returning them.

And btw, I don't mean add a series of if/else's. Take a look at the SubjectServiceController that accepts like 6 or so parameters, so adding that many if/else's just makes it hard to read. I recommend using a filter if it's not null, like I have done inside the controller. Because if we ever added more params e.g. geolocation, it's just another filter instead of playing around with if/else's.